### ISSUE TITLE
plasma 5

### ISSUE THREAD

**Issue by tomtomme** [felixhummel/deltrack#2]
this used to work fine with plasma5, maybe because I still had old kde4 stuff in the system. After a new OS (openSUSE Tumbleweed) installation today I am not able to make this work again via shortcut. What works is the alternate installation method, so I can click a button in the main menu and that works.
But pressing the selected shortcut (F10) does not. And the normal installation method complains, when I want to import the inputactions.hotkey file and says that it is empty...
is this something new in plasma5?
could you adapt the script?
thx tomme

**Comment by felixhummel** [comment]
Sure, I'll have a look.
This could take a while though, because I need to reproduce this (and I'm still on Kubuntu 14.04).

The shortcut should work though... Have you tried binding some other script to the shortcut and see if it works (e.g. `echo 'foo' > /tmp/foo`)?

**Comment by tomtomme** [comment]
sorry, my bash / scripting skills are very very basic. I do not know how to do that - I am not even sure I understand what you want me to do...

**Comment by felixhummel** [comment]
no problem. Simply do the [Alternative Installation](https://github.com/felixhummel/deltrack#alternative-installation-with-launcher) but instead of setting the command to `python $HOME/bin/deltrack.py` set it to `echo 'foo' > /tmp/foo`.
Then press your shortcut and check if `/tmp/foo` exists.
a) it does, which means deltrack is broken
b) it does not, which means it's KDE's fault

**Comment by tomtomme** [comment]
thanks, that explains it. I can confirm that foo exists after hitting my hotkey (F10):

l /tmp/foo
-rw-r--r-- 1 tom users 0 21. Okt 18:12 /tmp/foo

Strange. Because deltrack can´t be entirely broken cause clicking the menu entry deletes a track.

**Comment by felixhummel** [comment]
OK, so we know that the shortcut triggers the command given.
Let's try to look at the error messages of deltrack:
Replace the command with `python $HOME/bin/deltrack.py 2>&1 > /tmp/deltrack.log` (with your path to deltrack.py of course).
With any luck `/tmp/deltrack.log` should then contain some useful error messages...

**Comment by tomtomme** [comment]
Ok, now it is getting really weired. I did this:

python $HOME/bin/deltrack.py 2>&1 > /tmp/deltrack.log

saved, and hit F10, but the produced log is empty:

cat /tmp/deltrack.log

gives nothing.

l /tmp/deltrack.log
-rw-r--r-- 1 tom users 0 25. Okt 08:57 /tmp/deltrack.log

l /home/tom/1data/build/deltrack.py
-rwxr-xr-x 1 tom users 53108 21. Okt 11:20 /home/tom/1data/build/deltrack.py*

**Comment by felixhummel** [comment]
True. It should give *something* at least.
What happens when you run `python deltrack.py` directly from console?
Any output there?

**Comment by tomtomme** [comment]
now we have at least something:
```
python deltrack.py
  File "deltrack.py", line 4
    <!DOCTYPE html>
    ^
SyntaxError: invalid syntax
```

**Comment by felixhummel** [comment]
Nice one! :D
You downloaded the wrong file (probably the github HTML page).
You need this one: https://raw.githubusercontent.com/felixhummel/deltrack/master/deltrack.py
(notice the `raw` at the beginning)

**Comment by tomtomme** [comment]
yeah what I did was "right click" and "save as". the thing is, that firefox then thinks this is a html file. still - the code worked by clicking on the deltrack menu entry - very very strange. so even if the file was wrapped in html it could delete tracks. while hitting f10 did not delete them.

so now I clicked on deltrack.py, then raw, then copy pasted the content and overwrote the old content.
now "python deltrack.py" and also "F10" work!!!

Thank you so much - this was such an idiot error on my site - wow.


however, you might want to know that:
```
python3 deltrack.py
  File "deltrack.py", line 20
    print "Could not connect to dbus."
                                     ^
SyntaxError: Missing parentheses in call to 'print'
```
which might be expected as python 3 and 2 are incompatible if I am not mistaken.
as this is another issue I´ll close this one :)
